pkg/csi/service/wcpguest/controller_helper.go

```diff
@@ -213,9 +215,82 @@ func getPersistentVolumeClaimSpecWithStorageClass(pvcName string, namespace stri
 			StorageClassName: &storageClassName,
 		},
 	}
+	snapshotApiGroup := common.VolumeSnapshotApiGroup
+	volumeSnapshotKind := common.VolumeSnapshotKind
+	if volumeSnapshotName != "" {
+		localObjectReference := &v1.TypedLocalObjectReference{
+			APIGroup: &snapshotApiGroup,
+			Kind:     volumeSnapshotKind,
+			Name:     volumeSnapshotName,
+		}
+		claim.Spec.DataSource = localObjectReference
+	}
 	return claim
 }
 
+func getVolumeSnapshotWithVolumeSnapshotClass(volumeSnapshotName string, namespace string,
```

Comment:
How about rename this to constructVolumeSnapshotWithVolumeSnapshotClass? "get" sounds like an API get.